repo.js

```diff
@@ -111,15 +121,19 @@ class Repo extends EventEmitter {
     subscription.pipe(processSubscription)
 
     const swarm = new WebrtcSwarm(signalhub)
-
+    if (this.opts.gateway) {
+      const signalws = SignalhubWS(appName, [].concat(this.opts.gateway))
+      const swWs = new WebrtcSwarm(signalws)
+      swWs.on('peer', (stream, id) => this._replicate(stream, id, 'WebRtc'))
+    }
     this.swarm = swarm
 
-    swarm.on('peer', (stream) => this._replicate(stream))
+    swarm.on('peer', (stream, id) => this._replicate(stream, id, 'WebRtc'))
   }
 
-  _replicate (stream) {
+  _replicate (stream, id, type) {
+    this.emit('peer', {id: id, type: type})
     pump(stream, this.archive.replicate({
-      sparse: true,
```

Comment:
I was thinking that it would be good to use sparse by default to reduce the bandwidth required to open an archive.

Plus, I think that the Internet Archive use case wouldn't work if it wasn't sparse since the peer would try to download the entire archive to their browser by default. 😅

Reply:
Does `archive.replicate` take sparse? I thought it was just set at the archive level. https://github.com/mafintosh/hyperdrive#var-archive--hyperdrivestorage-key-options

Reply:
Oh snap! I can't believe I had that wrong for over a year. 😂 Thanks for the correction!